lib/MusicBrainz/Server/Form/Admin/Attributes.pm

```diff
@@ -41,11 +42,20 @@ has_field 'free_text' => (
     type => 'Boolean',
 );
 
+has_field 'entity_type' => (
+    type => 'Select',
+);
+
 sub options_parent_id {
     my ($self, $model) = @_;
     return select_options_tree($self->ctx, $self->ctx->stash->{model}, accessor => 'name');
 }
 
+sub options_entity_type {
+    my ($self) = @_;
+    return map { $_ => $_ } sort { $a cmp $b } entities_with('collections');
```

Comment:
Apparently [this works with lists too](http://search.cpan.org/dist/HTML-FormHandler-0.40057/lib/HTML/FormHandler/Field/Select.pm) (elsewhere we use array refs).